api/controllers/game/spectate.js

```diff
@@ -1,17 +1,37 @@
 module.exports = async function (req, res) {
   try {
     const { gameId } = req.body;
-    const game = await gameService.populateGame({ gameId });
+    const promiseGame = await gameService.populateGame({ gameId });
+    const promiseSpectator = await userService.findUser({ userId: req.session.usr });
 
+    const [game, spectator] = await Promise.all([promiseGame, promiseSpectator]);
     if (game.status || game.players.length < 2) {
-      return res.badRequest({message: 'You can only spectate an ongoing game with two players'});
+      return res.badRequest({ message: 'You can only spectate an ongoing game with two players' });
     }
-
     // Subscribe socket to game
-    Game.subscribe(req, [ gameId ]);
+    Game.subscribe(req, [gameId]);
     req.session.spectating = gameId;
 
-    return res.ok(game);
+    return Promise.all([game, spectator])
+      .then(async function updateDataBase() {
```

Comment:
```suggestion
      .then(async function updateDatabase() {
```
Casing here is off. It's also weird here to use chaining when async await would likely fit the bill. Not entirely sure, but something like this (high level, of course) makes more sense to me:
```
// game and spectator are already set, so you probably don't need to await them again imo
try {
    await updateDatabase(...);
    await populateGame(...);
    await publishAndRespond(...);
    // You'll need to set fullGame before this from one of those calls
    return res.ok(fullGame);
} catch {
   return res.badRequest(err);
}
```

Reply:
This is a pro move